### PR TITLE
update packngo to 0.5.1

### LINKED ISSUE #38: Update to packngo 0.5.0
### What problem are you facing?

The latest packngo version, 0.5.0, includes API backend and Go binding changes.

The backend API:
* URL is changed to `api.equinix.com/metal/v1` 
* device, vlan, attachment `href` values will include additional components (/metal/v1) in the URL path

The bindings have:
* changed the NetworkType detection and setting
* added a force detach volumes parameter to Device Delete

### How could the Equinix Metal Crossplane Provider help solve your problem?

* Update packngo to 0.5.0
  * device.NetworkType becomes device.GetNetworkType
  * setting device.NetworkType="type" becomes client.DevicePorts.ConvertDevice(device, "type")
  * mocking device with NetworkType="type" is more complicated and must simulate the features of the ports and bonding that are expressed through https://github.com/packethost/packngo/blob/2360ca405eb08fdf596ed3684f78138f2dbf64fa/ports.go#L226-L310 and the related functions
  * update attachment ID parsing of the `href` string

### PR DESCRIPTION
**Pull request by displague**
Fixes #38

With this update, the provider will access the
"api.equinix.com/metal/v1" API instead of "api.packet.net"

Setting and getting NetworkType for a device is done more
explicility through DevicePort and helper functions.

Signed-off-by: Marques Johansson <mjohansson@equinix.com>

### Checklist

I have:
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Ensured this PR contains a neat, self documenting set of commits.
- [ ] Updated any relevant [documentation], [examples], or [release notes].

[documentation]: https://github.com/crossplane/crossplane/tree/master/docs
[examples]: https://github.com/crossplane/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplane/crossplane/tree/master/PendingReleaseNotes.md

### CODE REVIEW

File: apis/server/v1alpha2/device_types.go [hunk 1]
```diff
@@ -171,9 +175,11 @@ type DeviceParameters struct {
 	// +optional
 	Features map[string]string `json:"features,omitempty"`
 
-	// +immutable
-	// +optional
-	IPAddresses []packngo.IPAddressCreateRequest `json:"ipAddresses,omitempty"`
```
Comment (displague) [inline apis/server/v1alpha2/device_types.go:176]: `go generate` was choking on this, so I aliased it. The generated documentation and spec relies on xref to the packngo repo. I suspect we don't want that.

https://github.com/packethost/crossplane-provider-equinix-metal/pull/39/files#diff-43c690743c8e782001de93e8166b5888e858256723d702a90bbc8aa8d2e58e47R96

File: package/crossplane.yaml [hunk 2]
```diff
@@ -23,11 +23,12 @@ metadata:
...
...Ljc5bC0uNjgtMWEuODMuODMsMCwwLDAsLjU5LS44M2MwLS42MS0uNDYtLjg5LTEuMDktLjg5aC0xLjI1Wm0uNjUtMS4zN3YtLjc1aC41NmMuMjgsMCwuNDYuMTQuNDYuMzdzLS4xNi4zOC0uNDYuMzhaIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgtMjUzLjk2IC0zNDUuNjEpIi8+PHBhdGggY2xhc3M9ImNscy0yIiBkPSJNMjczLjcyLDQyOC4zN0gyNzZsMy42OSw1LjQ2LDMuNjktNS40NmgyLjI2djExLjU0SDI4My41di04LjI4bC0zLjgyLDUuNDVoLS4wN2wtMy44LTUuNDF2OC4yNGgtMi4wOVoiIHRyYW5zZm9ybT0idHJhbnNsYXRlKC0yNTMuOTYgLTM0NS42MSkiLz48cGF0aCBjbGFzcz0iY2xzLTIiIGQ9Ik0yOTAuMTQsNDI4LjM3aDl2MS44MWgtNi44NnYzaDYuMDhWNDM1aC02LjA4djMuMDloN3YxLjgxaC05LjA4WiIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoLTI1My45NiAtMzQ1LjYxKSIvPjxwYXRoIGNsYXNzPSJjbHMtMiIgZD0iTTMwNi4wOCw0MzAuMjVoLTMuODZ2LTEuODhoOS44NXYxLjg4aC0zLjg2djkuNjZoLTIuMTNaIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgtMjUzLjk2IC0zNDUuNjEpIi8+PHBhdGggY2xhc3M9ImNscy0yIiBkPSJNMzE4LjQzLDQyOC4yOGgybDUuMzMsMTEuNjNoLTIuMjVsLTEuMjMtMi43OGgtNS43M2wtMS4yNSwyLjc4SDMxMy4xWm0zLjA1LDcuMDUtMi4wOS00LjYyLTIuMDgsNC42MloiIHRyYW5zZm9ybT0idHJhbnNsYXRlKC0yNTMuOTYgLTM0NS42MSkiLz48cGF0aCBjbGFzcz0iY2xzLTIiIGQ9Ik0zMjkuMjEsNDI4LjM3aDIuMTN2OS43aDYuMzd2MS44NGgtOC41WiIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoLTI1My45NiAtMzQ1LjYxKSIvPjxyZWN0IGNsYXNzPSJjbHMtMiIgeT0iNjUuNTIiIHdpZHRoPSIxMDQuMDYiIGhlaWdodD0iMC41Ii8+PC9zdmc+
```
Comment (displague) [inline package/crossplane.yaml:31]: Added an icon for the service: https://gist.github.com/displague/163c0fdf2ff15c52a58a3df4020c193a#file-em-stacked-color-svg (the square one)

File: pkg/clients/device/device.go [hunk 3]
```diff
@@ -50,6 +50,7 @@ type Client interface {
 type PortsClient interface {
 	DeviceToNetworkType(string, string) (*packngo.Device, error)
 	DeviceNetworkType(string) (string, error)
+	ConvertDevice(*packngo.Device, string) error
```
Comment (displague) [inline pkg/clients/device/device.go:53]: There is a weird relationship here: https://github.com/packethost/packngo/pull/229/files#diff-2121eef3cf61b5f3a4c8e545d742307a5f22b20a5952f964ad5c41d1d252676bL323-R322

DeviceToNetworkType just calls ConvertDevice, I could likely remove ConvertDevice here, and the Mocking around it. `UpdatedInstanceNetworkType` is the test that depends on it most, and I don't think that test is using the mocking as intended (yet, still passing).

I'm going to clean this up before removing draft status on the PR.

File: apis/server/v1alpha2/device_types.go (hunk 1 above)
Comment (displague) [reply apis/server/v1alpha2/device_types.go:176]:
```
15:37:39 [ .. ] go generate darwin_amd64
angryjet: error: error loading packages using pattern ./...: /Users/marques/src/crossplane-provider-equinix-metal/apis/server/v1alpha2/zz_generated.deepcopy.go:195:13: (*in)[i].DeepCopyInto undefined (type packngo.IPAddressCreateRequest has no field or method DeepCopyInto)
exit status 1
apis/generate.go:26: running "go": exit status 1
15:37:44 [FAIL]
make: *** [go.generate] Error 1
```

File: pkg/clients/device/device.go (hunk 3 above)
Comment (displague) [reply pkg/clients/device/device.go:53]: Cleaned up. Mostly.

I left ConvertDevice and MockConvertDevice references in place with https://github.com/packethost/packngo/issues/230 in mind.

File: apis/server/v1alpha2/device_types.go (hunk 1 above)
Comment (displague) [reply apis/server/v1alpha2/device_types.go:176]: The IPAddresses type is now local and copying to packngo.IPAddressCreateRequest is done just before sending the device create EM API call.

**Review by hasheddan — APPROVED** [review]

**Pull request merged**